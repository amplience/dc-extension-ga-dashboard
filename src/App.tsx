import React from 'react';
import Routes from './Routes';
import { Router } from 'react-router-dom';
import {createBrowserHistory} from 'history';
import {LocalizationProvider} from '@material-ui/pickers';
import MomentUtils from '@material-ui/pickers/adapter/moment';

const browserHistory = createBrowserHistory();

function App() {
  return (
    <LocalizationProvider dateAdapter={MomentUtils}>
      <Router history={browserHistory}>
        <Routes />
      </Router>
    </LocalizationProvider>
  );
}

export default App;
