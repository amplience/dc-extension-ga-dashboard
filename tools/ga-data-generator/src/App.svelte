<script>
  window.ga =
    window.ga ||
    function () {
      (ga.q = ga.q || []).push(arguments);
    };
	ga.l = +new Date();

  let gaId = 'UA-182105724-1',
    hitType = 'event',
    eventCategory = 'Banners',
    eventAction = 'Click',
    eventLabel = '/shop/mens/sale',
    dimension1 = '',
    dimension2 = '',
    dimension3 = '';

  const submit = () => {
    ga('create', gaId, 'auto');
    ga('send', {
      hitType,
      eventCategory,
      eventAction,
      eventLabel,
      dimension1,
      dimension2,
      dimension3,
    });
  };
</script>

<style>
  main {
    text-align: center;
    padding: 1em;
    margin: 0 auto;
    align-items: center;
  }

  h1 {
    color: #ff3e00;
    text-transform: uppercase;
    font-size: 4em;
    font-weight: 100;
  }

  .content {
    margin: 0 auto;
    width: 500px;
    display: grid;
    grid-template-columns: 30% 70%;
    grid-column-gap: 10px;
  }
</style>

<main>
  <h1>Google Analytics Data Generator</h1>
  <form class="content">
    <label>GA Tracking Code</label>
    <input type="text" bind:value={gaId} />

    <label>Hit Type</label>
    <input type="text" bind:value={hitType} />

    <label>Event Category</label>
    <input type="text" bind:value={eventCategory} />

    <label>Event Action</label>
    <input type="text" bind:value={eventAction} />

    <label>Event Label</label>
    <input type="text" bind:value={eventLabel} />

    <label>Dimension1 (ContentItemId)</label>
    <input type="text" bind:value={dimension1} />

    <label>Dimension2 (EditionId)</label>
    <input type="text" bind:value={dimension2} />

    <label>Dimension3 (SlotId)</label>
    <input type="text" bind:value={dimension3} />
  </form>
  <button on:click={submit}>Submit</button>
</main>
