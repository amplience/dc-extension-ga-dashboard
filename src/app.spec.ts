import { render } from '@testing-library/svelte';
import App from './app.svelte';

const mockGetClient = jest.fn();

jest.mock(
  './services/management-sdk-proxy/management-sdk-proxy.service',
  () => {
    return {
      default: jest.fn().mockImplementation(() => {
        return {
          hubId: 'hub-id',
          getClient: mockGetClient,
        };
      }),
    };
  }
);
jest.mock(
  './services/message-event-channel/message-event-channel.factory',
  () => {
    return {
      default: jest.fn().mockImplementation(() => ({ emit: () => undefined })),
    };
  }
);

describe('App', () => {
  afterEach(() => {
    jest.restoreAllMocks();
    jest.resetAllMocks();
  });

  beforeEach(() => {
    const mockSearchIndexesList = jest.fn().mockImplementation(() => {
      return {
        getItems: () => [],
      };
    });
    const mockHubsGet = jest.fn().mockImplementation(() => {
      return {
        related: {
          searchIndexes: {
            list: mockSearchIndexesList,
          },
        },
      };
    });
    mockGetClient.mockImplementation(() => {
      return {
        hubs: {
          get: mockHubsGet,
        },
      };
    });
  });

  it('should render the App component', () => {
    const { container } = render(App, {});

    // this only renders the loading state - we need more tests to cover loaded scenarios
    expect(container.firstChild).toMatchSnapshot();
  });
});
