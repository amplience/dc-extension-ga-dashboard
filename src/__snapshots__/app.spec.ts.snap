// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the App component 1`] = `
<div>
  <main
    class="svelte-fc0sjg"
  >
    <div
      class="top-bar svelte-fc0sjg"
    >
      <section
        class="top-bar svelte-1y25f5"
      >
        <div
          class="container svelte-1y25f5"
        >
          <section
            class="svelte-17fo8w9"
          >
            <div
              class="select-filter svelte-17fo8w9"
            >
              <span
                class="icon-wrapper svelte-17fo8w9"
              >
                <div
                  class=" svelte-zpbvvv"
                  style="--icon-width: 20px; --icon-height: 20px;"
                >
                  <object
                    src="ic-filter.svg/"
                  />
                </div>
              </span>
               
              <span
                class="svelte-17fo8w9"
              >
                No filters applied
              </span>
            </div>
             
          </section>
        </div>
      </section>
       
      <section
        class="date-range-bar svelte-e877a4"
      >
        <div
          class="container svelte-e877a4"
        >
          <span
            class="svelte-e877a4"
          >
            Date range
          </span>
           
           
          <section
            class=" svelte-ws099h"
          >
            <div
              class="date-range-inputs svelte-ws099h"
              data-testid="date-range-inputs"
            >
              <div
                class="date-selection svelte-ws099h"
              >
                <div
                  class="calendar-icon svelte-zpbvvv"
                  style="--icon-width: 24px; --icon-height: 24px;"
                >
                  <object
                    src="ic-view-calendar.svg/"
                  />
                </div>
                 
                <input
                  class="svelte-ws099h"
                  placeholder="Pick a start date"
                  readonly=""
                />
              </div>
               
              <div
                class="date-selection svelte-ws099h"
              >
                <div
                  class="calendar-icon svelte-zpbvvv"
                  style="--icon-width: 24px; --icon-height: 24px;"
                >
                  <object
                    src="ic-view-calendar.svg/"
                  />
                </div>
                 
                <input
                  class="svelte-ws099h"
                  placeholder="Pick an end date"
                  readonly=""
                />
              </div>
            </div>
             
          </section>
        </div>
      </section>
    </div>
     
    <div
      class="scroll-area svelte-fc0sjg"
    >
      <section
        class="container svelte-fc0sjg"
      >
        <div
          class="spinner-container svelte-wkqaim"
          style="--z-index: 10;"
        >
          <div
            class="spinner dot-0 svelte-wkqaim"
          />
          <div
            class="spinner dot-1 svelte-wkqaim"
          />
          <div
            class="spinner dot-2 svelte-wkqaim"
          />
        </div>
      </section>
    </div>
    <iframe
      aria-hidden="true"
      src="about:blank"
      style="display: block; position: absolute; top: 0; left: 0; width: 100%; height: 100%; overflow: hidden; border: 0; opacity: 0; pointer-events: none; z-index: -1;"
      tabindex="-1"
    />
  </main>
</div>
`;
